File: step-timetable-backend/src/main/java/org/tyaa/itstep/dashboard/models/TimeIntervalModel.java
package org.tyaa.itstep.dashboard.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Builder;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.math.BigInteger;
import java.util.ArrayList;
import java.util.List;

@Data
@Builder(toBuilder = true)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
@Document
public class TimeIntervalModel implements Serializable {
    @Id
    private String id;
    private Integer pairNumber;
    private String intervalStart;
    private String intervalEnd;
    private List<LessonModel> lessons = new ArrayList<>();
}

File: step-timetable-frontend/src/models/TimeIntervalModel.ts
import LessonCardModel from './LessonCardModel'

export default class TimeIntervalModel {
  // private static lastId = 0
  public id?: number
  public pairNumber: number
  public intervalStart: string
  public intervalEnd: string
  public lessonCards: LessonCardModel[]
  constructor (
      intervalStart: string,
      intervalEnd: string,
      lessonCards: LessonCardModel[] = [],
      pairNumber: number,
      id?: number
    ) {
    if (id) {
      this.id = id
    }/*  else {
      this.id = ++TimeIntervalModel.lastId
    } */
    this.pairNumber = pairNumber
    this.intervalStart = intervalStart
    this.intervalEnd = intervalEnd
    this.lessonCards = lessonCards
  }
}